src/extensions/registries/page-registry.ts

```diff
@@ -20,12 +24,29 @@ export interface PageComponents {
   Page: React.ComponentType<any>;
 }
 
+const routePrefix = "/extension/:name"
+
+export function getPageUrl(ext: LensExtension, baseUrl = "") {
+  const validUrlName = ext.name.replace("@", "").replace("/", "-");
+  return compile(routePrefix)({ name: validUrlName }) + baseUrl;
+}
+
 export class PageRegistry<T extends PageRegistration> extends BaseRegistry<T> {
-  getItems() {
-    return super.getItems().map(item => {
-      item.routePath = item.extension.getPageRoute(item.routePath)
-      return item
-    });
+
+  @action
+  add(items: T[], ext?: LensExtension) {
+    const normalizedItems = items.map((i) => {
+      i.routePath = getPageUrl(ext, i.routePath)
+      return i
+    })
+    return super.add(normalizedItems);
+  }
+
+  getByPageMenuTarget(target: PageMenuTarget) {
+    if (!target) {
+      return null
+    }
+    return this.getItems().find((page) => page.routePath.startsWith(`/extension/${target.extensionId}/`) && page.id === target.pageId)
```

Comment:
Btw I guess `exact` param should participate in this check?

Reply:
Why is that? It's just trying to find matching page (Lens itself generates this prefix so it should be safe for matching).